### PR TITLE
Settings page

### PR DESCRIPTION
**Pull request by cvergaray**
This pull request introduces a new Settings Management page, a UI for creating and modifying the settings JSON. While the JSON can be hand crafted ahead of time, now spinning up a new instance of Label_web is easier because you create the instance and adjust all the settings from within the tool itself.
A new configuration management module was added to improves how printer defaults and configuration errors are handled and displayed in the application. It also adds related frontend enhancements and new tests to ensure correct behavior when dealing with printer defaults and configuration validation.

**Key changes include:**

### Configuration Management Enhancements

* Added the Settings page for managing the configuration of the application from the GUI. This is backwards compatible with previous versions of the configuration json, but also adds a couple of new options.
* The printer include/exclude list and the enabled sizes options added to the config. Both of these options are to limit what is displayed in the UI, either white/black list-ing printers and exclusively enabling media. If no options are provided, then the application works as it did before, showing everything.
* Added a new `configuration_management.py` module that centralizes configuration handling, including loading, saving, validation, format conversion, and utility functions for label sizes and printers. This improves maintainability and consistency across the application.
* Enhanced configuration validation logic to allow operation when the configured printer is missing but CUPS provides a default, and to provide clearer error messages for various misconfiguration scenarios.

### Improved Printer Default Handling and Error Tracking

* Enhanced the initialization logic in `implementation_cups.py` to automatically use the CUPS-reported default printer if no printer is specified in the config, and to track initialization errors for better diagnostics. Also, improved error handling when fetching printers and label sizes from CUPS. [[1]](diffhunk://#diff-48f9a235a71885aa2029df28de9dfe9312949749d5e2e5fa1680608e7b622e6aR39-R71) [[2]](diffhunk://#diff-48f9a235a71885aa2029df28de9dfe9312949749d5e2e5fa1680608e7b622e6aR233-R247) [[3]](diffhunk://#diff-48f9a235a71885aa2029df28de9dfe9312949749d5e2e5fa1680608e7b622e6aL363-R397)

### Frontend User Experience Improvements

* Added a configuration errors indicator to the navigation bar and styled it for both desktop and mobile views, making configuration issues more visible to users. [[1]](diffhunk://#diff-e0344cdf6b6d4ea40aecb37dc475caed5919e76497756615aa47e5ae40a8a02dR18-R44) [[2]](diffhunk://#diff-8abaccbf8bda76d9756d1ec97c5c4efdebc34043e8d521880195a3991e47aa74L26-R37) This indicator appears if configuration errors are detected, guiding users to the settings page for resolution. [[1]](diffhunk://#diff-8abaccbf8bda76d9756d1ec97c5c4efdebc34043e8d521880195a3991e47aa74L26-R37) [[2]](diffhunk://#diff-e0344cdf6b6d4ea40aecb37dc475caed5919e76497756615aa47e5ae40a8a02dR18-R44)
* Introduced a new `static/js/common.js` with utility functions for escaping HTML, loading printer media, and checking configuration errors, supporting more interactive and robust frontend behavior.

### Testing

* Added `tests/test_printer_defaults.py` to verify correct handling of printer defaults and configuration validation, ensuring that the application behaves as expected in various edge cases.

This pull request introduces significant improvements to configuration management, error handling, and user interface feedback for configuration issues, along with new utility functions and tests for printer defaults. The main focus is on making configuration handling more robust, surfacing errors to users, and improving maintainability through modularization and testing.

### CODE REVIEW

File: views/settings.jinja2 [hunk 1]
```diff
@@ -0,0 +1,1802 @@
...
+          errorDiv.textContent = 'Width and height must be positive numbers (pixels).';
+          errorDiv.style.display = 'block';
+          return false;
+        }
+        return true;
+      }
+
+      // If no explicit dimensions, key must be parseable
+      // Check for patterns like "4x6in", "100x50mm", etc.
+      const parseablePattern = /^(\d+(?:\.\d+)?)\s*x\s*(\d+(?:\.\d+)?)\s*(in|mm|cm)?$/i;
```
[the remark this answers is not in the document]
Comment (cvergaray) [reply views/settings.jinja2:692]: Added comment to indicate that the two must be kept in sync. I tried setting it once and injecting into the js, but I ran into issues with the formatting, as JS and Python use different regex syntaxes.

File: views/labeldesigner.jinja2 [hunk 2]
```diff
@@ -53,9 +53,11 @@
               <div class="chooser panel-body">
               <label for="fontFamily">Font Family:</label>
               <select class="form-control" id="fontFamily" onChange="preview()">
+                {% set default_family = label.get('DEFAULT_FONTS', {}).get('family', '') %}
+                {% set default_style = label.get('DEFAULT_FONTS', {}).get('style', '') %}
```
[the remark this answers is not in the document]
Comment (cvergaray) [reply views/labeldesigner.jinja2:57]: If there are those many failures and the worst thing that happens is the default font doesn't get set, then the user can still select which font they want or it will default to the first in the list. I am not too concerned with this scenario.
